### DISCUSSION TITLE
meshlabserver in latest release disappeared!

### DISCUSSION THREAD

**Discussion by Richi90** [cnr-isti-vclab/meshlab#897] [Q&A] [answered]
Hello guys, 
I've automatically updated Meshlab 2020.03 to the latest 2020.12. 
However, I've noticed that meshlabserver executable disappeared from the .zip folder.

Is there anything that I've missing?
Can I replace the executable from the oldest version without problems?

Thanks in advance.
Riccardo

**Comment by alemuntoni** [comment, marked as answer]
See https://github.com/cnr-isti-vclab/meshlab/issues/882 and https://github.com/cnr-isti-vclab/meshlab/discussions/843.

**Reply by Richi90** [reply]
Hi, thanks for the quick reply. I've missed those threads.
Unfortunately, I don't have time to see/learn the new software that replaces meshlabserver. 
I've also all the codes already written and ready to be used, sadly.
I'll move back to the previous version.
Thanks again!
Grazie!
Riccardo

**Reply by jmespadero** [reply]
Hope this helps, I think we should implement an meshlabserver.py script to give same functionality than old meshlabserver.

Usage : python3 meshlabserver.py inputMesh outputMesh [filter.mlx]

```python 
import pymeshlab as ml
import sys 
ms = ml.MeshSet()
ms.load_new_mesh(sys.argv[1])
if len(sys.argv) > 3 :
  #Load and apply an optative .mlx filter. 
  ms.load_filter_script(sys.argv[3])
  ms.apply_filter_script()
ms.save_current_mesh(sys.argv[2])

```

**Reply by cignoni** [reply]
Very good idea. We should add this!

**Reply by jmespadero** [reply]
@cignoni @alemuntoni  I can write a not-so-simple script so one can apply several mlx filters and choose if normal, textures, should be added to output meshes. 
I would like to hear your oppinion about complety mimic old meshlabserver parameters (you must to write "-i" to input filenames , and "-o" to output, etc...) or create a new simplified parameter system (if there is two arguments, assume first is input and second is output).

**Reply by alemuntoni** [reply]
@jmespadero I agree that it is a very good idea, but I think that it would also be temporary..
Ideally, in next meshlab versions I would like to drop mlx scripts in favour of python scripts that use pymeshlab directly. 
I still don't know when this will be available, but I hope soon. Therefore, I would keep things simple. A script like the one you wrote here would be perfect!
